lib/auth/auth_with_roles.go

```diff
@@ -5069,6 +5052,49 @@ func (a *ServerWithRoles) GetAppServersAndSAMLIdPServiceProviders(ctx context.Co
 	return appsAndSPs, nil
 }
 
+func (a *ServerWithRoles) checkAccessToSAMLIdPServiceProvider(samlSP types.SAMLIdPServiceProvider) error {
+	return a.context.Checker.CheckAccess(
+		samlSP,
+		// MFA is not required for operations on SAML IdP service provider resources but
+		// will be enforced at the connection time.
+		services.AccessState{MFAVerified: true})
+}
+
+// GetSAMLIdPServiceProviders returns all registered SAMLIdPServiceProviders resources.
+// Results are filtered with RBAC.
+func (a *ServerWithRoles) GetSAMLIdPServiceProviders(ctx context.Context, namespace string, pageSize int32, sortBy types.SortBy) ([]types.SAMLIdPServiceProvider, error) {
```

Comment:
At first glance this API doesn't make sense, it looks like we pass it a limit in the request

https://github.com/gravitational/teleport/blob/736dc28029c4ab770acff799d1254d68dc089d15/lib/auth/auth_with_roles.go#L1835

however we don't respect that limit, we simply gather all the `SAMLIdPServiceProvider`s and return them.

Reply:
Good point. That is a result of trying to shoehorn a paginated API within GetAll op. I tried to avoid a default guesswork and pass in whatever is passed in request but I guess that just is more confusing. I'll have it fixed.

Reply:
https://github.com/gravitational/teleport/pull/39119/commits/917a2b5c709e3802204193834df26496a87dbc96